app/models/concerns/aliquotable.rb

```diff
@@ -34,5 +34,23 @@ module Aliquotable
     has_many :derived_aliquots, -> { where(aliquot_type: :derived) },
              as: :source, class_name: 'Aliquot',
              dependent: :nullify, inverse_of: :source
+
+    # Method to calculate the used volume
+    def used_volume
+      derived_aliquots.sum(&:volume)
+    end
+
+    # Method to calculate the available volume
+    def available_volume
+      primary_aliquot.volume - used_volume
+    end
+
+    # Method to check if there is enough volume available
+    def volume_check(required_volume)
+      return true if available_volume >= required_volume
```

Comment:
@stevieing @BenTopping  I thought it make sense to return true in cases where the required_volume is equal to available_volume. Something for you to confirm.

Reply:
It might be that we need to provide a message that the volume is exhausted but we don't know until it gets used.